finality-aleph/src/justification/requester.rs

```diff
@@ -0,0 +1,97 @@
+use std::{collections::VecDeque, marker::PhantomData, time::Duration};
+
+use log::{info, warn};
+use tokio::time::sleep;
+
+use crate::{
+    justification::LOG_TARGET,
+    network::RequestBlocks,
+    session::SessionBoundaryInfo,
+    sync::{ChainStatus, Header, Justification},
+    BlockIdentifier,
+};
+
+pub struct Requester<J, RB, CS>
+where
+    J: Justification,
+    RB: RequestBlocks<<J::Header as Header>::Identifier> + 'static,
+    CS: ChainStatus<J>,
+{
+    block_requester: RB,
+    chain_status: CS,
+    boundary_info: SessionBoundaryInfo,
+    _phantom: PhantomData<J>,
+}
+
+// This should only be necessary during the rolling update, so lets make the request period
+// ridiculously long, so that it doesn't introduce too much weird behaviour otherwise.
+const REQUEST_PERIOD: Duration = Duration::from_secs(60);
+
+impl<J, RB, CS> Requester<J, RB, CS>
+where
+    J: Justification,
+    RB: RequestBlocks<<J::Header as Header>::Identifier> + 'static,
+    CS: ChainStatus<J>,
+{
+    pub fn new(block_requester: RB, chain_status: CS, boundary_info: SessionBoundaryInfo) -> Self {
+        Requester {
+            block_requester,
+            chain_status,
+            boundary_info,
+            _phantom: PhantomData,
+        }
+    }
+
+    fn children_ids(
+        &self,
+        id: <J::Header as Header>::Identifier,
+    ) -> Result<VecDeque<<J::Header as Header>::Identifier>, CS::Error> {
+        Ok(self
+            .chain_status
+            .children(id)?
+            .into_iter()
+            .map(|header| header.id())
+            .collect())
+    }
+
+    fn perform_request(&self) -> Result<(), CS::Error> {
+        self.block_requester.clear_justification_requests();
```

Comment:
Is this call necessary? `extra_requests.rs` seems to handle cleaning well. If it's necessary, a comment here would be good idea.

Reply:
Oh, and it definitely won't happen more than once per minute.